src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProjectProvider.cs

```diff
@@ -92,6 +90,8 @@ public sealed class LegacyPackageReferenceProjectProvider : INuGetProjectProvide
         {
             var componentModel = await _componentModel.GetValueAsync();
 
+            await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
This method casts to VSProject4 which is a STA bound API. 

The initialization of the IComponentModel is freethreaded, so no need to switch to the UI before that.